php/php.blade/src/org/netbeans/modules/php/blade/editor/BladeBracesMatcher.java

```diff
@@ -0,0 +1,288 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.php.blade.editor;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.swing.text.BadLocationException;
+import org.antlr.v4.runtime.Token;
+import org.netbeans.api.annotations.common.NonNull;
+import org.netbeans.api.editor.mimelookup.MimeRegistration;
+import org.netbeans.editor.BaseDocument;
+import org.netbeans.modules.php.blade.syntax.BladeDirectivesUtils;
+import org.netbeans.modules.php.blade.syntax.BladeTagsUtils;
+import org.netbeans.modules.php.blade.syntax.antlr4.v10.BladeAntlrLexer;
+import static org.netbeans.modules.php.blade.syntax.antlr4.v10.BladeAntlrLexer.*;
+import org.netbeans.modules.php.blade.syntax.antlr4.v10.BladeAntlrUtils;
+import org.netbeans.spi.editor.bracesmatching.BracesMatcher;
+import org.netbeans.spi.editor.bracesmatching.BracesMatcherFactory;
+import org.netbeans.spi.editor.bracesmatching.MatcherContext;
+
+/**
+ * brace matcher - block directives : @if @endif .. - output echo statements {{
+ * }} {!! !!}
+ *
+ * @author bogdan
+ */
+public class BladeBracesMatcher implements BracesMatcher {
+
+    public enum BraceDirectionType {
+        END_TO_START, START_TO_END, CUSTOM_START_TO_END, CURLY_END_TO_START, CURLY_START_TO_END, STOP
+    }
+    private final MatcherContext context;
+    private Token originToken;
+    private BraceDirectionType currentDirection;
```

Comment:
Probably, we should set an initial value (`BraceDirectionType.STOP`? or add `NONE`?).

Reply:
Setting a default constant value of BraceDirectionType.NONE;